pkg/kubelet/eviction/eviction_manager.go

```diff
@@ -496,23 +502,43 @@ func (m *managerImpl) emptyDirLimitEviction(podStats statsapi.PodStats, pod *v1.
 			}
 		}
 	}
+
+	return false
+}
+
+func (m *managerImpl) podEphemeralStorageLimitEviction(podStats statsapi.PodStats, pod *v1.Pod) bool {
+	_, podLimits := apiv1resource.PodRequestsAndLimits(pod)
+	_, found := podLimits[v1.ResourceEphemeralStorage]
+	if !found {
+		return false
+	}
+
+	podEphemeralStorageTotalUsage := &resource.Quantity{}
+	podEphemeralStorageTotalUsage.Add(*getTotalEmptyDirUsed(podStats, pod))
+	podEphemeralStorageTotalUsage.Add(*getTotalContainerEphemeralStorageUsed(podStats))
```

Comment:
To get the total storage usage for a pod, please also check podDiskUsage() in helpers.go